### PR TITLE
`1.0.0` release

### PR DESCRIPTION
**Pull request by ZeroAct**
- reduce functions
- add test codes

### CODE REVIEW

**Review by dustjr1565 — APPROVED** [review]

File: waffle_utils/file/search.py [hunk 1]
```diff
@@ -1,99 +1,73 @@
...
-    if isinstance(extension, str):
-        extension = [extension]
-    elif extension is None:
-        extension = []
+    files = directory.glob(f"**/*" if recursive else "*")
+    files = list(filter(lambda file: file.is_file(), files))
+    if extension:
+        if isinstance(extension, str):
+            extension = [extension]
+        files = [file for file in files if file.suffix.lower() in extension]
```
Comment (Jiyong-Jeon) [inline waffle_utils/file/search.py:35]: When we receive an extension, there is no guarantee that we will receive it in lowercase.

I think we should change the extension to lowercase (`lower()`) within the function.

**Review by Jiyong-Jeon — COMMENTED** [review]
LGTM

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -12,33 +12,26 @@ jobs:
...
       - name: Check environment
         run: |
           python --version
           pip --version
           pip list
       - name: Pytest tests
-        shell: bash  # for Windows compatibility
-        run: python -m pytest tests
+        shell: bash
+        run: python -m pytest tests/ -v
```
[the remark this answers is not in the document]
Comment (ZeroAct) [reply .github/workflows/ci.yaml:37]: ok

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+@pytest.fixture(scope="session")
+def dummy_text(tmpdir_factory):
+    directory = Path(tmpdir_factory.mktemp("dummy_text_file"))
+    dummy_file = Path(directory) / "dummy_text_file.txt"
+    data = "dummy_text_file"
+    with open(dummy_file, "w") as f:
+        f.write(data)
+
+    return {"path": dummy_file, "data": data}
```
[the remark this answers is not in the document]
Comment (ZeroAct) [reply tests/conftest.py:26]: ok

File: waffle_utils/file/search.py (hunk 1 above)
Comment (ZeroAct) [reply waffle_utils/file/search.py:35]: ok

Comment (ZeroAct) [reply waffle_utils/file/search.py:35]: good

**Review by Jiyong-Jeon — APPROVED** [review]

**Pull request merged**